src/main/java/org/jenkinsci/plugins/authorizeproject/AuthorizeProjectStrategy.java

```diff
@@ -53,5 +59,35 @@
      * @param item the item in queue, which will be a build.
      * @return
      */
-    public abstract Authentication authenticate(AbstractProject<?, ?> project, Queue.Item item);
+    public Authentication authenticate(Job<?, ?> project, Queue.Item item) {
+        if(!Util.isOverridden(
+                AuthorizeProjectStrategy.class,
+                getClass(),
+                "authenticate",
+                AbstractProject.class,
+                Queue.Item.class
+        )) {
+            throw new AbstractMethodError();
+        }
+        
+        if (!(project instanceof AbstractProject)) {
+            Descriptor<?> d = Jenkins.getInstance().getDescriptor(getClass());
+            LOGGER.log(
+                    Level.WARNING,
+                    "This authorization strategy ({0}) is designed for authorize-project < 1.1.0 and not applicable for non-AbstractProjects (like WorkflowJob). ignored.",
+                    (d != null)?d.getDisplayName():getClass().getName()
+            );

```

Comment:
E.g. [Ownership Plugin](https://wiki.jenkins-ci.org/display/JENKINS/Ownership+Plugin) implements the old `AuthorizeProjectStrategy` and results in following system logs when you use that with `WorkflowJob`:
```
5 06, 2015 12:11:06 PM org.jenkinsci.plugins.authorizeproject.AuthorizeProjectStrategy authenticate
WARNING: This authorization strategy (Run as Job Owner) is designed for authorize-project < 1.1.0 and not applicable for non-AbstractProjects (like WorkflowJob). ignored.
```

I once tried throwing an exception and making build failed, but gave up for an useless and confusing result: builds are not started and no logs are left.